### PR TITLE
Refactoring Variables and Callstack

### PR DESCRIPTION
**Pull request by KsavinN**

### CODE REVIEW

File: src/variables/body/variablesSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import React, { useState, useRef } from 'react';
+
+import useOutsideClick from './useOutsideClick';
+
+import { Variables } from '../index';
+
+export class VariablesSearch extends Widget {
+  constructor(model: any) {
+    super();
+    this.addClass('jp-DebuggerVariables-Search');
```
Comment (afshin) [inline src/variables/body/variablesSearch.tsx]: `jp-DebuggerVariables-Search` -> `jp-DebuggerVariables-search` (lowercase)

File: src/variables/body/variablesSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Jupyter Development Team.
```
Comment (afshin) [inline src/variables/body/variablesSearch.tsx]: I sould suggest renaming this file to `search.tsx` and renaming the class to `class Search extends Widget`.

File: src/variables/body/variablesBody.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Jupyter Development Team.
```
Comment (afshin) [inline src/variables/body/variablesBody.ts]: I would suggest just calling these `Body` and `Description` and renaming the file to `index.ts`

File: src/variables/body/useTbody.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Jupyter Development Team.
```
Comment (afshin) [inline src/variables/body/useTbody.tsx]: Is this function used in multiple places? If not, it should probably just be defined in the file where it's used. I'm not sure I understand why it's not in-line where you call it.

File: src/variables/body/variablesSearch.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  const changeScope = (newScope: string) => {
+    if (newScope === scope) {
+      return;
+    }
+    setScope(newScope);
+    setToggle(false);
+  };
+
+  const List = (
+    <ul className="jp-MenuComponent">
```
Comment (afshin) [inline src/variables/body/variablesSearch.tsx]: `jp-MenuComponent` name is too generic. You might not even need it, though. You might be able to do:
`.jp-DebuggerVariables ul`

File: src/variables/body/variablesSearch.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    </div>
+  );
+};
+
+class VariableScopeSearch extends ReactWidget {
+  constructor() {
+    super();
+    this.node.style.overflow = 'visible';
+    this.node.style.width = '100px';
+    this.addClass('jp-SearchScope');
```
Comment (afshin) [inline src/variables/body/variablesSearch.tsx]: Similarly this should probably just be `class ScopeSearch` and `jp-DebuggerVariables-scope`

File: src/variables/body/variablesSearch.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+class VariableSearchInput extends ReactWidget {
+  search: string;
+  model: Variables.IModel;
+  constructor(model: Variables.IModel) {
+    super();
+    this.model = model;
+    this.search = model.filter;
+    this.node.style;
+    this.addClass('jp-SearchInput');
```
Comment (afshin) [inline src/variables/body/variablesSearch.tsx]: This class name is too generic, `jp-DebuggerVariables-input` maybe?

**Comment by afshin** [comment]
I think we should merge this iteration. Thanks @KsavinN!

**Comment by KsavinN** [comment]
Merged to Master [no conflicts]

**Pull request closed without merging**